forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -171,223 +220,107 @@ def _monte_carlo_dfe(program: Program, qubits: Sequence[int], in_states: list, n
-        pauli_var_est=var_est,
-        fid_point_est=fid['expectation'],
-        fid_var_est=fid['variance'],
-        fid_std_err_est=np.sqrt(fid['variance'])
-    )
-
-
-def aggregate_process_dfe(data: dict):
-    if isinstance(data['dimension'], int):
-        dim = data['dimension']
+    exp = TomographyExperiment(list(
+        _monte_carlo_dfe(program=program, qubits=qubits,
+                         in_states=[None, plusX, minusX, plusY, minusY, plusZ, minusZ],
+                         n_terms=n_terms, benchmarker=benchmarker)),
+        program=program, qubits=qubits)
+    return DFEexperiment(exp,'process')
+
+
+def acquire_dfe_data(qc: QuantumComputer, dfe: DFEexperiment, n_shots = 10_000, active_reset=False):
+    # TODO: add number of shots
+    #       add variance bounds?
+    res = list(measure_observables(qc, dfe.exp, n_shots=n_shots, active_reset=active_reset))
+    return DFEdata(res, 
+                   in_states = [str(exp[0].in_state) for exp in dfe.exp],
+                   program = dfe.exp.program,
+                   out_pauli = [str(exp[0].out_operator) for exp in dfe.exp], 
+                   pauli_point_est = np.array([r.expectation for r in res ]), 
+                   pauli_var_est = np.array([r.stddev**2 for r in res]), 
+                   cal_point_est = np.array([r.calibration_expectation for r in res ]), 
+                   cal_var_est = np.array([r.calibration_stddev**2 for r in res ]), 
+                   dimension = 2**len(dfe.exp.qubits), 
+                   qubits = dfe.exp.qubits,
+                   type = dfe.type)
+
+
+def analyse_dfe_data(data: DFEdata):
+    mean = np.sum([r.expectation for r in data.res])
+    variance = np.sum([r.stddev**2 for r in data.res])
+    d = data.dimension
+
+    if data.type == 'state':
+        mean_est = np.mean([r.expectation for r in data.res])
+        var_est = np.sum([r.stddev**2 for r in data.res])/len(data.res)**2
+    elif data.type == 'process':
+        mean_est = (d**2 * np.mean([r.expectation for r in data.res]) + d)/(d**2+d)
+        v = np.sum([r.stddev**2 for r in data.res])/len(data.res)**2
+        var_est = d**2/(d+1)**2 * v
     else:
-        dim = list(set(data['dimension']))
-        assert len(dim) == 0
-        dim = dim[0]
-    return {
-        'expectation': (1.0 + np.sum(data['expectation']) + dim) / (dim * (dim + 1)),
-        'variance': np.sum(data['variance']) / (dim * (dim + 1)) ** 2,
-    }
-
+        raise ValueError('DFEdata can only be of type \'state\' or \'process\'.')
 
-def aggregate_state_dfe(data: dict):
-    return {
-        'expectation': (1.0 + np.sum(data['expectation'])) / data['dimension'],
-        'variance': np.sum(data['variance'])
-    }
+    return DFEestimate( dimension= data.dimension,
+                        qubits = data.qubits,
+                        fid_point_est = mean_est, 
+                        fid_var_est = var_est, 
+                        fid_std_err_est = np.sqrt(var_est))
```

Comment:
here it is `std_err_est` will have to go through the whole repo...

Reply:
It really should be std error everywhere, no? We are talking about estimates.

Reply:
Yes, I was just pointing out it was done correctly here but not up there^^.